src/Compilers/CSharp/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -361,5 +361,20 @@ private static bool IsDeconstructionCompatibleArgument(ExpressionSyntax expressi
 
             return false;
         }
+
+        internal static SimpleNameSyntax? GetInterceptableNameSyntax(this InvocationExpressionSyntax invocation)
+        {
+            // If a qualified name is used as a valid receiver of an invocation syntax at some point,
+            // we probably want to treat it similarly to a MemberAccessExpression.
+            // However, we don't expect to encounter it.
+            Debug.Assert(invocation.Expression is not QualifiedNameSyntax);
+
+            return invocation.Expression switch
+            {
+                MemberAccessExpressionSyntax memberAccess => memberAccess.Name,
```

Comment:
what about `?.M()` or `->M()`?

Reply:
This is moving pre-existing logic. Filed #72478 to follow up on these scenarios.